draft-ietf-dprive-dnsoquic.md

```diff
@@ -326,7 +342,12 @@ SERVFAIL.
 
 If a server is incapable of sending a DNS response due to an internal error, it
 may issue a QUIC Stream Reset with error code DOQ_INTERNAL_ERROR. The
-corresponding transaction MUST be abandoned.
+corresponding transaction MUST be abandoned. Clients MAY limit the number of
+DOQ_INTERNAL_ERROR errors received on a connection before choosing to close the
+connection.
+
```

Comment:
Maybe add mention that if the stream is abandoned because a client sent a stop sending request, the error should be DOQ_REQUEST_CANCELLED.

Also, it is technically possible for clients to open a stream, send the first bytes of a query on it, then issue a stream reset. I suppose the server will have to reset the stream in that case.

Reply:
Not really. The application errors are there to ease debugging, I would not use them for any kind of enforcement. The point is that stop_sending and reset_streams can be abused for abnormal behavior, for example to implement the slowloris attack described in [section 21.6. Slowloris Attacks of RFC 9000](https://www.rfc-editor.org/rfc/rfc9000.html#name-slowloris-attacks). So applications may take defensive actions if they detect abusive behavior. (MT will say this is obvious...)

Reply:
I would write, "Clients MAY limit the number of unsolicited
QUIC Stream Reset received on a connection before choosing to close the
connection." Or something like that. Point is that the error code does not matter. Obviously, clients should not worry if the stream reset after they sent a stop sending request.